Objects/genobject.c

```diff
@@ -327,7 +327,7 @@ _PyGen_yf(PyGenObject *gen)
     PyObject *yf = NULL;
     PyFrameObject *f = gen->gi_frame;
 
-    if (f && f->f_stacktop) {
+    if (f) {
```

Comment:
Would it make sense to replace the `&& f->f_stacktop` with `&& f->f_stackdepth >=0` or at least an `assert(f->f_stackdepth >= 0)` on line 329? 

Admittedly, I'm a bit out of my depth when it come to the low-level generator details, but at a glance it seems to me like we would not want to start with the yield from with a stack depth of -1 (or lower).

Reply:
In theory, the stack depth can never be negative. There's an assertion on line 344 that it is positive.

Reply:
Yeah, I saw that assertion on line 344. If it's realistically not going to be negative, we probably only need the assertion prior to modifying the stack depth (particularly for decrements).